tests/keras/preprocessing/test_image.py

```diff
@@ -90,6 +90,23 @@ def test_img_flip(self):
                 assert ((potentially_flipped_x == x).all() or
                         (potentially_flipped_x == flip_axis(x, col_index)).all())
 
+    def test_flow_index(self):
+        N = 11
+        batch_size = 3
+        seed = 0
+        np.random.seed(seed)
+        idx = np.random.permutation(N)
+
+        generator = ImageDataGenerator()
+        flow_index = generator._flow_index(N, batch_size=batch_size, shuffle=True, seed=seed)
+        for i in range(0, N, batch_size):
+            index_array = next(flow_index)[0]
+            if N < i + batch_size:
+                assert generator.batch_index == 0
+            else:
+                assert generator.batch_index == (i // batch_size) + 1
+            assert (index_array == idx[i:i+batch_size]).all()
```

Comment:
The random permutation `idx` is a completely different random permutation than the one produced internally by the generator.   

Shouldn't you just test that all indices are covered exactly once, i.e. that the union of all `index_arrays` after ciel(N/batch_size) is  [N-1]?